Tensile/Tests/unit/test_HardwarePredicates.py

```diff
@@ -82,8 +108,13 @@ def test_hardware_library_merge_order(libraries):
         lib.merge(lib2)
 
     assert lib.rows[-1]['predicate'] == HardwarePredicate('TruePred')
-    assert lib.rows[0]['predicate'] == HardwarePredicate.FromHardware((9,0,8), 64)
-    assert lib.rows[1]['predicate'] == HardwarePredicate.FromHardware((9,0,8), 60)
+    # assert lib.rows[0]['predicate'] == HardwarePredicate.FromHardware((9,0,8), 64)
+    # assert lib.rows[1]['predicate'] == HardwarePredicate.FromHardware((9,0,8), 60)
+    assert lib.rows[0]['predicate'] == HardwarePredicate.FromHardware((9,4,2), isAPU=1)
```

Comment:
Could you explain a bit about why the order is in this way?

Reply:
The library sort order prioritizes specialized tunings over generic ones so that those kernels get picked before the fallback ones.
If an APU mode is set then those libs are sorted first. (IsAPU=1 is APU, IsAPU=0 is XPU, IsAPU=None or -1 is generic).
If a CU count is set then those libs are sorted next, with higher CU counts taking priority (in practice we don't have specialized libs with different CU counts so this is used as "set or not set").
Then everything else is sorted by architecture.
The code technically supports both APU mode and CU count being set, but there are no plans to use that option so ordering there doesn't matter much. But if we did use it, current priority goes to APU mode then CU count, but we could revise if a use-case came up.